src/sickle.ts

```diff
@@ -200,29 +196,31 @@ class Annotator {
     let lastEnd = node.getFullStart();
     for (let child of node.getChildren()) {
       let childStart = child.getFullStart();
-      this.writeSourceBefore(lastEnd, child);
+      this.writeTextFromOffset(lastEnd, child);
       this.visit(child);
       lastEnd = child.getEnd();
     }
     // Write any trailing text.
-    let text = node.getSourceFile().getText().slice(lastEnd, node.getEnd());
-    if (text) this.emit(text);
+    this.writeRange(lastEnd, node.getEnd());
   }
 
-  private writeTextBetween(node: ts.Node, to: ts.Node) {
-    let text = node.getSourceFile().getText().slice(node.getFullStart(), to.getFullStart());
+  // Write a span of the input file as expressed by absolute offsets.
+  // These offsets are found in attributes like node.getFullStart() and
+  // node.getEnd().
+  private writeRange(from: number, to: number) {
+    let text = this.file.text.slice(from, to);
```

Comment:
IMHO still needs a comment explaining why `getSourceFile().getText()` is wrong here.